concourse/pipelines/create-bosh-concourse.yml

```diff
@@ -655,7 +655,7 @@ jobs:
 
               terraform apply \
                 -auto-approve=true \
-                -var concourse_egress_cidr="${CONCOURSE_EGRESS_IP}/32" \
```

Comment:
I think this is code may be working by accident, or maybe just in a mildly confusing way?

You're setting `concourse_egress_cidr` to be an array with a single CIDR block, but you're not wrapping it in quotes within, so it'd be invalid JSON: `[1.2.3.4/32]`.

It seems like Terraform is happy to parse it all the same, because later on you're using this expression

`var.concourse_egress_cidrs == [] ? [] : formatlist("%s/32", data.aws_instances.concourse_workers.public_ips)`

If the variable is empty, use the empty array, otherwise pull the CIDR ranges from the Terraform data source. 

It's ignoring any values that were passed in.

If we intend to ignore the values being passed in, we should remove the variable altogether.

Reply:
Although the values of the variable are ignored, setting the variable in earlier jobs and then not setting the variable in the tasks within the expunge-concourse job is the device that is used to remove the ip addresses of the concourse workers from the security groups to which they were earlier added.

Reply:
Aha, gotcha! Maybe a more intuitive, intent-revealing approach would be to set it to null when expunging?

Reply:
Yes, agreed.

Reply:
I wasn't able to set the concourse_egress_cdrs variable to null or [] in the expunge-concourse job in accordance with Andy's last comment. I found an explanation [here](https://github.com/hashicorp/terraform/issues/29078) as to why null doesn't work but I could not find anything similar for setting an empty list on the command line. I discussed this with Andy and it was agreed that I would update the README.md with that explanation. Tom spotted that concourse workers can be added within  the concourse-deploy job and then used in the post-deploy step. I have created a new job - add-new-workers-to-bosh - to ensure that if new workers are added in the concourse-deploy job they do have access to bosh if used in the subsequent post-deploy job.